backend/coreapp/github.py

```diff
@@ -0,0 +1,111 @@
+from django.conf import settings
+from django.core.cache import cache
+from django.db import models, transaction
+from django.contrib.auth import login
+from django.contrib.auth.models import User
+from rest_framework import status
+from rest_framework.exceptions import APIException
+
+from typing import Optional
+from github import Github
+from github.NamedUser import NamedUser
+from .middleware import Request
+import requests
+
+from .models import Profile
+
+API_CACHE_TIMEOUT = 60 * 60 # 1 hour
+
+class BadOAuthCode(APIException):
+    status_code = status.HTTP_401_UNAUTHORIZED
+    default_detail = "Invalid or expired GitHub OAuth verification code."
+    default_code = "bad_oauth_code"
+
+class MissingOAuthScope(APIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    default_code = "bad_oauth_scope"
+
+    def __init__(self, scope: str):
+        super(f"The GitHub OAuth verification code was valid but lacks required scope '{scope}'.")
+
+class GitHubUser(models.Model):
+    user = models.OneToOneField(
+        User,
+        on_delete=models.CASCADE,
+        primary_key=True,
+        related_name="github",
+    )
+    github_id = models.PositiveIntegerField(unique=True, editable=False)
+    access_token = models.CharField(max_length=100)
+
+    class Meta:
+        verbose_name = "GitHub user"
+        verbose_name_plural = "GitHub users"
+
+    def details(self, use_cache: bool = True) -> NamedUser:
```

Comment:
I don't think the `use_cache` parameter is ever used? (is there a `.details(use_cache=False)` missing somewhere?)